internal/generate/types_test.go

```diff
@@ -111,48 +246,24 @@ func Test_createTypeObject(t *testing.T) {
 				Value: &openapi3.Schema{Type: &openapi3.Types{"string"}, Format: "uuid"},
 			},
 			"type": {
-				Value: &openapi3.Schema{Type: &openapi3.Types{"string"}, Enum: []interface{}{"snapshot"}},
+				Value: &openapi3.Schema{Type: &openapi3.Types{"string"}, Enum: []any{"snapshot"}},
 			},
 		}}
 
-	type args struct {
-		s        openapi3.Schema
-		name     string
-		typeName string
-	}
-	tests := []struct {
-		name string
-		args args
-		want TypeTemplate
-	}{
-		{
-			name: "success",
-			args: args{typesSpec, "DiskSource", "DiskSourceSnapshot"},
-			want: TypeTemplate{
-				Description: "Create a disk from a disk snapshot\n//\n// Required fields:\n// - Type",
-				Name:        "DiskSourceSnapshot",
-				Type:        "struct", Fields: []TypeField{
-					{
-						Description:       "",
-						Name:              "SnapshotId",
-						Type:              "string",
-						SerializationInfo: "`json:\"snapshot_id,omitempty\" yaml:\"snapshot_id,omitempty\"`",
-					},
-					{
-						Description:       "",
-						Name:              "Type",
-						Type:              "DiskSourceType",
-						SerializationInfo: "`json:\"type\" yaml:\"type\"`",
-					},
-				},
-			},
+	got := createTypeObject(&typesSpec, "DiskSource", "DiskSourceSnapshot", "Create a disk from a disk snapshot")
+
+	want := TypeTemplate{
+		Name:        "DiskSourceSnapshot",
+		Type:        "struct",
+		Description: "Create a disk from a disk snapshot\n//\n// Required fields:\n// - Type",
+		Fields: []TypeField{
+			{Name: "SnapshotId", Type: "string", MarshalKey: "snapshot_id", Required: false},
+			{Name: "Type", Type: "DiskSourceType", MarshalKey: "type", Required: true},
 		},
 	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			got := createTypeObject(&tt.args.s, tt.args.name, tt.args.typeName, "Create a disk from a disk snapshot")
-			assert.Equal(t, tt.want, got)
-		})
+
+	if diff := cmp.Diff(want, got, cmpIgnoreSchema); diff != "" {
```

Comment:
This is an example of an assertion we could replace with a comparison of want/got generated code, rather than comparing the intermediate representations. In this case, I don't have a strong preference about which style is better. WDYT @lgfa29?

Reply:
I think something more like an end-to-end test sounds good, but maybe we'd have a smaller number of bigger tests there. Most of the tests here are making assertions about a small surface area, like whether we're generating a description or type correctly for given inputs, and I don't know that we'd want to generate a large number of those tests as openapi specs and desired output files. So maybe I'll leave these tests alone for now and think about adding end-to-end tests later.